server/chat-plugins/net-filters.ts

```diff
@@ -309,14 +320,35 @@ export const commands: ChatCommands = {
 			}
 			const targetPath = FS(`logs/chat/${roomid}/${date.slice(0, -3)}/${date}.txt`);
 			if (!targetPath.existsSync()) return this.errorReply(`Logs for that date not found`);
-			this.sendReply(`Initating training...`);
-			const response = await PM.query({data: {path: targetPath.path, result}, type: 'trainfrom'});
+			this.privateGlobalModAction(`${user.name} used /netfilter trainfrom ${roomid} (${date})`);
+			const response = await NeuralNetChecker.query({data: {path: targetPath.path, result}, type: 'trainfrom'}, true);
 			this.sendReply(`Training completed in ${Chat.toDurationString(response[0])}`);
 			this.privateGlobalModAction(
 				`${user.name} trained the net filters on logs from ${roomid} (${date} - ${Chat.count(response[1], 'lines')})`
 			);
 			this.stafflog(`Result: ${result} | Time: ${response[0]}ms`);
 		},
+		cancel: 'stop',
+		stop(target, room, user) {
+			checkAllowed(this);
+			let count = 0;
+			if (!PM.releasingProcesses.length) {
+				return this.errorReply(`No train tasks are pending`);
+			}
+			for (const subProcess of PM.releasingProcesses) {
+				for (const [task, resolve] of subProcess.pendingTasks) {
+					resolve({error: `The pending train query was cancelled.`});
```

Comment:
This is all the releasing processes, not just pending train queries, though.

Reply:
Yeah, that's intentional - i know to watch for the processes to be careful, and i know most releasing processes are the train ones usually. No other process should be releasing except during hotpatch, and those would release much quicker.

Reply:
It seems like this is bad just as a correctness thing. It seems like training should be a different PM if you want to do stuff like this.